lib/go/thrift/simple_server.go

```diff
@@ -149,8 +151,10 @@ func (p *TSimpleServer) Serve() error {
 }
 
 func (p *TSimpleServer) Stop() error {
-	p.quit <- struct{}{}
-	p.serverTransport.Interrupt()
+	if atomic.CompareAndSwapInt64(&p.stopped, 0, 1) {
```

Comment:
Hi, still looking for a reply to my last comment about 0 and 1.  I think your code relies on the fact that stop is called twice, as it will skip the first bump of the counter from 0 to 1, and then on the second call to stop it will actually stop, so I cannot approve the request.  If I misread it please let me know.